source/view.c

```diff
@@ -1302,13 +1302,15 @@ static void selection_changed_user_callback(unsigned int index, RofiViewState *s
   if (config.on_selection_changed == NULL)
     return;
 
-  static int last_index = -1;
+  static unsigned int last_index = UINT32_MAX;
```

Comment:
I prefer not to have "hidden" state.  If there is no way around this variable, it should be part of  `RofiViewState` .

Reply:
I agree it's a bit dirty, will try to find a cleaner solution otherwise I'll add a new field in RofiViewState.